### PR TITLE
feat(reporting): add faithful XLSX renderer for the ReportModel

### PR DESCRIPTION
**Pull request by Noah231515**
## What

Adds `render.XLSX` — the **second renderer** over the engine's `ReportModel`, and the faithful, spreadsheet-native counterpart to the flat CSV renderer.

## Layout (dimension-columns / pivot)

The group-by dimensions are leading columns (each value shown **once per group**), the report columns follow, and subtotal/grand-total rows carry a `Total` marker in the column at the group's depth (the "staircase"). As it appears in Excel:

```
Paid By | Tag  | Category | Count | Subtotal | Hst  | Total  | Avg
Dana    | Alex | Clothing |   4   | 120.00   | 15.60| 135.60 | 33.90
        |      | Medical  |   2   |  80.00   |  0.00|  80.00 | 40.00
        |      | Total    |   6   | 200.00   | 15.60| 215.60 | 35.93
        | Sam  | ...      |       |          |      |        |
        | Total|          |  10   | 320.00   | 27.30| 347.30 | 34.73
Grand Total    |          |  10   | 320.00   | 27.30| 347.30 | 34.73
```

## Spreadsheet-native touches

- **Native, typed number cells** with a number format (not strings), so the workbook is analyzable. Currency defaults to `#,##0.00` (or `ColumnDescriptor.Format` / `Meta.CurrencyFormat` if set); this is where the engine's opaque `Format` hint is finally consumed.
- **Bold** header, subtotal, and grand-total rows.
- Sheet name `Report`.

## Scope

- **Static values, not formulas.** Translating arithmetic columns into live `=SUM`/expression cell formulas — the reason `ColumnDescriptor.Expr` is an exported AST — is a deliberate later slice. Document chrome/slots (title, logo) await the template work.
- **Pure downstream consumer** of the `ReportModel` — fetches and computes nothing, touches no engine/pipeline/model code (there is no layout machinery upstream; layout is entirely the renderer's).
- Shares the `Dimension` type and the `groupBy`-depth guard (extracted as `validateGroupByDepth`) with the CSV renderer, but **not** the walk (flat vs faithful are genuinely different layouts).
- Adds `github.com/xuri/excelize/v2` (pure Go; CGo-free, so CI is unaffected).

## Tests

Models are built via `reporting.Run`, rendered, then the bytes are **round-tripped back through `excelize.OpenReader`** to assert:
- the full display grid (structure, dimension blanking, the subtotal staircase, grand total, currency display),
- native number storage (a number's raw value differs from its formatted display — a text cell wouldn't),
- the currency number format, and bold on header/subtotal/grand-total but not detail rows.

Case matrix parallels the CSV suite: sums + aggregates with the non-linear AVG rolling up per level; each aggregate function (SUM/COUNT/AVG/MIN/MAX); nested groups + staircase + blanking; records mode with a multi-value label; no-grouping; subtotals-on/grand-off; the `(None)` bucket; multi-value fan-out; empty report; a group-depth mismatch error (shared guard); and unicode.

No orchestrator or HTTP handler in this slice — the renderer produces bytes and is unit-tested in isolation, consistent with the engine, `ReportDataService`, and the CSV renderer.

## Summary by CodeRabbit

* **New Features**
  * Added XLSX report export with grouped detail rows, subtotals, and optional grand totals.
  * Spreadsheet output uses native numeric cells (with number/currency formatting) and supports Unicode text and empty reports.
  * Total rows and headers are bold, with consistent label handling.
* **Bug Fixes**
  * Improved group-by depth validation for rendering consistency (including CSV output).
* **Documentation**
  * Expanded XLSX rendering documentation with layout and static-value behavior.
* **Tests**
  * Added comprehensive XLSX rendering test coverage for totals, edge cases, and formatting.

### CODE REVIEW

**Comment by Noah231515** [comment]
@coderabbit review

File: api/internal/reporting/render/xlsx.go [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+			return nil, err
+		}
+	}
+
+	buffer, err := writer.file.WriteToBuffer()
+	if err != nil {
+		return nil, err
+	}
+	return buffer.Bytes(), nil
+}
```
[the remark this answers is not in the document]
Comment (Noah231515) [reply api/internal/reporting/render/xlsx.go:69]: **ACCEPT** — added `defer writer.file.Close()` immediately after constructing the `excelize.File` (commit 3a94bf5b).

Agreed it isn't a live leak on this write-only `NewFile()` path (`Close()` only clears temp files created by read operations, of which there are none here), but the defer is the idiom in excelize's own examples, costs nothing, and guards against a future streaming/file-open change. Verified safe: the deferred close runs *after* `WriteToBuffer()` has produced the bytes, and the returned `bytes.Buffer` is independent of the file's internal temp state, so the output is unaffected — the 12 round-trip tests stay green.
